pkg/security/secl/model/process_cache_entry.go

```diff
@@ -59,6 +59,21 @@ func (pc *ProcessCacheEntry) GetNextAncestorNoFork() *ProcessCacheEntry {
 	return ancestor
 }
 
+// HasCompleteLineage returns false if, from the entry, we cannot ascend the ancestors list to PID 1
+func (pc *ProcessCacheEntry) HasCompleteLineage(maxDepth int) bool {
+	if pc.Pid == 1 { // we reach pid 1, the lineage is complete
+		return true
+	} else if pc.Ancestor == nil {
+		return false
+	}
+
+	if maxDepth == 0 {
+		return false
+	}
+
+	return pc.Ancestor.HasCompleteLineage(maxDepth - 1)
```

Comment:
nit:
maxDepth checks seems uncessary (as we never fail)
reuse this function in functional tests

Reply:
Indeed, failure is not an option. Guard removed